### PR TITLE
Bump stackage resolver and make package multi-target

### PR DESCRIPTION
**Pull request by onslaughtq**
This library has been updated to target GHC 7.10.3
and GHC 8.6.5, so that upstream dependencies such
as data-shipping, and eventually NCIS, can use it
wihtout forcing us to upgrade other apps, such
as schmods, immediately.

### CODE REVIEW

**Comment by onslaughtq** [comment]
I've removed the `test-all-logs` directory.  

The dependencies for Happstack list a base version equal to 4.*, so it's not too surprising that it works between these different LTS versions.  Is there a specific test you want me to try besides just compiling?

**Comment by jbrechtel** [comment]
No, just surprised. happstack-server has all kinds of other dependencies, too, that must vary greatly between the two vastly different LTS versions. I expect that it will work if it compiles and I don't have any suspicions about specific things that might not work if it did compile.

File: src/Data/Validation/Types/Pure.hs [hunk 1]
```diff
@@ -51,24 +51,41 @@ mapErrors f (Invalid errs) = Invalid (f errs)
...
   mempty = Messages Set.empty
+  mappend = errorsAppend
+
+errorsAppend :: Errors
+             -> Errors
+             -> Errors
+errorsAppend (Messages m) (Messages m') = Messages (m `mappend` m')
+errorsAppend (Group g) (Group g')       = Group (Map.unionWith mappend g g')
+errorsAppend g m@(Messages _)           = g `mappend` nestErrors "" m
+errorsAppend m g                        = nestErrors "" m `mappend` g
```
Comment (qxjit) [inline src/Data/Validation/Types/Pure.hs]: Not all that important, but I probably would have named this `appendErrors`

File: src/Data/Validation/Types/Pure.hs [hunk 2]
```diff
@@ -51,24 +51,41 @@ mapErrors f (Invalid errs) = Invalid (f errs)
...
+  mappend = validationResultAppend
+
+validationResultAppend :: Monoid m
+                       => ValidationResult m
+                       -> ValidationResult m
+                       -> ValidationResult m
+validationResultAppend (Valid a) (Valid a')     = Valid (a `mappend` a')
+validationResultAppend (Invalid e) (Invalid e') = Invalid (e `mappend` e')
+validationResultAppend (Valid _) invalid        = invalid
+validationResultAppend invalid (Valid _)        = invalid
```
Comment (qxjit) [inline src/Data/Validation/Types/Pure.hs]: Similarly, I would have named this `appendValidationResult` (or `...Results`)

File: src/Data/Validation/XML.hs [hunk 3]
```diff
@@ -119,12 +120,20 @@ parseBool text =
...
 
 instance Monoid Content where
   mempty = Content ""
+  mappend = contentAppend
+
+contentAppend :: Content
+              -> Content
+              -> Content
+contentAppend (Content t) (Content t') = Content (t `mappend` t')
+contentAppend _ _                      = NotContent
```
Comment (qxjit) [inline src/Data/Validation/XML.hs]: Same here

**Pull request merged**